lib/semantics/expression.cc

```diff
@@ -1079,10 +1079,10 @@ void ArrayConstructorContext::Add(const parser::AcValue &x) {
                 std::get<parser::AcImpliedDoControl::Bounds>(control.t)};
             Analyze(bounds.name);
             parser::CharBlock name{bounds.name.thing.thing.source};
+            const Symbol *symbol{bounds.name.thing.thing.symbol};
             int kind{IntType::kind};
-            if (auto &its{std::get<std::optional<parser::IntegerTypeSpec>>(
```

Comment:
If an explicit integer type appears in the parse tree, it should be used rather than applying implicit typing to the symbol, yes?

Reply:
Yes, but the symbol type resolution in:
https://github.com/flang-compiler/f18/blob/8c63aa94972fb3f7b88f23fa1b78f8b67ffa8732/lib/semantics/resolve-names.cc#L4153
is already doing all this, so I do not see a good reason to re-do it here.

The last line I added in the test file is actually checking that explicit integer types are still applied in here.